<?xml version="1.0" encoding="utf-8"?>
<!--
Launch file for Selective ICP with rviz
-->

<launch>

  <!-- Proper Output Info -->
  <env name="ROSCONSOLE_FORMAT" value="[${severity}] (${node}) [${time}]: ${message}\n" />

  <param name="/use_sim_time" value="true" />
    <!--<arg name="cad_file" default="$(find cpt_selective_icp)/resources/cla.json" />-->
    <arg name="rosbag" />
    <arg name="logprefix" />
    <arg name="segmentation_model" />
    <arg name="input_height" />
    <arg name="input_width" />
    <arg name="threshold" default="150" />

    <node pkg="background_foreground_segmentation"
          type="inference_node.py"
          name="inference_node"
          output="screen">
        <param name="model_gdrive_id" value="$(arg segmentation_model)" />
        <param name="input_height" value="$(arg input_height)" />
        <param name="input_width" value="$(arg input_width)" />
        <rosparam>
          image_topics:
            - /pickelhaube/cam0/image
            - /pickelhaube/cam1/image
            - /pickelhaube/cam2/image
            - /pickelhaube/cam3/image
          segmentation_output_topics:
            - /pickelhaube/cam0/segmented
            - /pickelhaube/cam1/segmented
            - /pickelhaube/cam2/segmented
            - /pickelhaube/cam3/segmented
          pointcloud_topic: /rslidar_points
        </rosparam>
    </node>

    <node pkg="background_foreground_segmentation"
          type="cloud_filtering_node"
          name="cloud_filter"
          output="screen">
        <param name="cloud_topic" value="/rslidar_points" />
        <param name="cloud_frame" value="rslidar" />
        <param name="filtered_pc_topic" value="/points_filtered" />
        <param name="density_threshold" value="$(arg threshold)" />
        <rosparam param="image_topics">
            - /pickelhaube/cam0/segmented
            - /pickelhaube/cam1/segmented
            - /pickelhaube/cam2/segmented
            - /pickelhaube/cam3/segmented
        </rosparam>
        <rosparam param="caminfo_topics">
            - /pickelhaube/cam0/camera_info
            - /pickelhaube/cam1/camera_info
            - /pickelhaube/cam2/camera_info
            - /pickelhaube/cam3/camera_info
        </rosparam>
    </node>

    <!---->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find background_foreground_segmentation)/config/segmented_icp.rviz" />

    <!-- should be started last for handover to ICP-->
    <include file="$(find background_foreground_segmentation)/launch/alphasense_tools/pickelhaube_from_bag.launch">
        <arg name="rosbag" value="$(arg rosbag)" />
    </include>

</launch>
